### PR TITLE
Add support for --pathmap

### LINKED ISSUE #5213: Signature data embedded in executable contains absolute source file path
The `FSharpSignatureData` resource embedded in the assembly contains absolute paths to the source files. This could include a user's real name etc.
I know this info is in the pdb, but shipping that with an executable is optional.
Executables produced from C# compiler don't contain file paths

#### Repro steps

Provide the steps required to reproduce the problem

1. Create a simple `test.fs` containing, say, `let main argv = 1`
2. `fsc test.fs`
3. `strings test.exe`
4. observe the full path to `test.fs`

#### Expected behavior

No source paths in the exe

#### Actual behavior

Absolute paths are embedded in the `FSharpSignatureData` resource

#### Known workarounds

Provide a description of any known workarounds.

#### Related information 

Provide any related information 

* Windows
* Microsoft (R) F# Compiler version 10.1.0 for F# 4.1
* .NET Runtime

### PR DESCRIPTION
**Pull request by saul**
Fixes #5213, #3812

Supersedes #5801

### CODE REVIEW

**Comment by saul** [comment]
cc @brettfo can you take a look at this PR instead?

File: src/fsharp/CompileOps.fs [hunk 1]
```diff
@@ -2149,6 +2149,8 @@ type TcConfigBuilder =
       mutable internalTestSpanStackReferring: bool
 
       mutable noConditionalErasure: bool
+      
+      mutable pathMap: PathMap
```
Comment (brettfo) [inline src/fsharp/CompileOps.fs]: nit: the order here (`noConditionalErasure, pathMap`) doesn't match the order used below (`pathMap, noConditionalErasure`)

**Comment by KevinRansom** [comment]
@saul 

conflict ...

/help tests fail, baseline updates probably.
````
CompilerOptions\fsc\help (-?) -- failed
CompilerOptions\fsc\help (--help) -- failed
CompilerOptions\fsc\help (/?) -- failed

CompilerOptions\fsi\help (-?-40) -- failed
CompilerOptions\fsi\help (--help-40) -- failed
CompilerOptions\fsi\help (/?-40) -- failed
CompilerOptions\fsi\help (-? --nologo-40) -- failed

CompilerOptions\fsi\exename (-?-40) -- failed

A determinism test:
CompilerOptions\fsc\determinism (20) -- failed
````

**Comment by saul** [comment]
Thanks @KevinRansom - I've just updated the baselines.

The determinism test (which I haven't changed) that failed is potentially flaky. In the `env.lst`, there is a comment stating: "we need to use the timeout because non-determinism in fsc is due to datetime, so we need to guarentee at least 1 sec difference"

...but there is no timeout/pause to ensure this gap as far as I can see?

**Comment by saul** [comment]
Re-running tests as it looks like `Optimizations\Inlining (Match01.fs)` is flaky (it also failed in a previous run). It looks like in some places, the lambda has changed name. For example, `CompareTo$cont@4` vs `CompareTo$cont@4-53`.

@dsyme - any idea why the compiler is sometimes choosing different names? It actually seems important to determinism more widely.

**Comment by saul** [comment]
Looks like a different flaky test now:

```
Failed   Automation.EnumDUInterfacefromFSBrowse.InsideMatch
Error Message:
 System.IO.IOException : The process cannot access the file 'D:\a\1\s\artifacts\tmp\Release\b9916641-ec81-40f0-b656-238061953256\testFile.fs' because it is being used by another process.
Stack Trace:
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.File.InternalDelete(String path, Boolean checkHost)
   at System.IO.File.Delete(String path)
   at <StartupCode$VisualFSharp-UnitTests>.$ProjectOptionsBuilder.System-IDisposable-Dispose@32.Invoke(Tuple`2 tupledArg) in D:\a\1\s\vsintegration\tests\UnitTests\ProjectOptionsBuilder.fs:line 32
   at Microsoft.FSharp.Primitives.Basics.List.iter[T](FSharpFunc`2 f, FSharpList`1 x) in D:\a\1\s\src\fsharp\FSharp.Core\local.fs:line 91
   at VisualFSharp.UnitTests.Roslyn.FSharpProject.System-IDisposable-Dispose() in D:\a\1\s\vsintegration\tests\UnitTests\ProjectOptionsBuilder.fs:line 32
   at Microsoft.VisualStudio.FSharp.Editor.Tests.Roslyn.QuickInfo.GetQuickInfoTextFromCode(String code) in D:\a\1\s\vsintegration\tests\UnitTests\QuickInfoTests.fs:line 45
   at Microsoft.VisualStudio.FSharp.Editor.Tests.Roslyn.QuickInfo.Automation.EnumDUInterfacefromFSBrowse.InsideMatch() in D:\a\1\s\vsintegration\tests\UnitTests\QuickInfoTests.fs:line 95
```

**Comment by cartermp** [comment]
It's green! Let's not sneeze in the wrong direction here, for fear of angering another flaky test...

**Pull request merged**